Fixing SerDe for Worker class, and fixing issue with SerDe for defaultdicts

the-elancier opened the pull request:
## Description
- Fixing SerDe for Worker class
- Also fixed an issue with not being able to deserialize defaultdict

Closes https://github.com/OpenMined/Heartbeat/issues/19#issue-1563772479


## How has this been tested?
Tested serializing and deserializing the worker class

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

the-elancier commented:
@madhavajay @shubham3121 Regarding the fixing of SerDe for defaultdict, please review it and let me know if this fix is alright. If there's a better way to do this, please do let me know and I will modify accordingly. Thanks.

madhavajay reviewed (approved):
Great work! 🙌🏽

I have enabled the new tests and fixed a few small issues which opened up some more code to the serde and quickly fixed that as well.

For the defaultdict that was a pretty solid start. I changed it to propagate the type as well and now i think it works for all cases.

Excellent work.

the-elancier commented:
> Great work! 🙌🏽
> 
> I have enabled the new tests and fixed a few small issues which opened up some more code to the serde and quickly fixed that as well.
> 
> For the defaultdict that was a pretty solid start. I changed it to propagate the type as well and now i think it works for all cases.
> 
> Excellent work.

@madhavajay  Thanks. Your fix for defaultdicts is definitely much more robust. Amazing work.
